pubsubgrpc.go

```diff
@@ -1,69 +1,54 @@
-package libpubsubgrpc
+package pubsubgrpc
 
 import (
 	"context"
-	"net"
-	"time"
-
-	"sync"
-
 	"io"
+	"time"
 
-	"github.com/RTradeLtd/go-libp2p-pubsub-grpc/pb"
+	pb "github.com/RTradeLtd/go-libp2p-pubsub-grpc/pb"
 	"github.com/libp2p/go-libp2p-core/host"
 	discovery "github.com/libp2p/go-libp2p-discovery"
 	ps "github.com/libp2p/go-libp2p-pubsub"
 	"go.uber.org/zap"
-	"google.golang.org/grpc"
 )
 
-// Server is used to run the libp2p pubsub grpc server
-// it enables communicating pubsub messages over grpc
-type Server struct {
-	pb pb.PubSubServiceServer
+// Service represents a pubsub object
+type Service struct {
+	dev bool
+
 	ps *ps.PubSub
 	sd *discovery.RoutingDiscovery
 	h  host.Host
+
+	l *zap.SugaredLogger
 }
 
-// NewServer is used to intiialize a pubsub grpc server and run it
-func NewServer(ctx context.Context, wg *sync.WaitGroup, pubsub *ps.PubSub, sd *discovery.RoutingDiscovery, h host.Host, logger *zap.SugaredLogger, insecure bool, protocol, url string) error {
-	lis, err := net.Listen(protocol, url)
-	if err != nil {
-		return err
-	}
-	var serverOpts []grpc.ServerOption
-	if !insecure {
-		serverOpts, err = options("", "", "", logger)
-		if err != nil {
-			return err
-		}
-	}
-	srv := &Server{ps: pubsub, sd: sd, h: h}
-	gServer := grpc.NewServer(serverOpts...)
-	pb.RegisterPubSubServiceServer(gServer, srv)
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		for {
-			select {
-			case <-ctx.Done():
-				logger.Info("shutting server down")
-				gServer.GracefulStop()
-				return
-			}
-		}
-	}()
-	return gServer.Serve(lis)
+// NewService creates and returns a new service
+func NewService(
+	dev bool,
+	ps *ps.PubSub,
+	sd *discovery.RoutingDiscovery,
+	h host.Host,
+	logger *zap.SugaredLogger,
+	secured bool,
+	protocol,
+	url string,
+) (*Service, error) {
+	return &Service{
+		dev: dev,
+		ps:  ps,
+		sd:  sd,
+		h:   h,
+	}, nil
 }
 
-// GetTopics is used to return a list of all known topics the pubsub instance is subscribed to.
-func (s *Server) GetTopics(ctx context.Context, req *pb.Empty) (*pb.TopicsResponse, error) {
+// GetTopics returns a TopicsResponse structure
+func (s *Service) GetTopics(ctx context.Context, req *pb.Empty) (*pb.TopicsResponse, error) {
 	return &pb.TopicsResponse{Names: s.ps.GetTopics()}, nil
 }
 
-// ListPeers is used to return a list of peers subscribed to a given topic or topics
-func (s *Server) ListPeers(ctx context.Context, req *pb.ListPeersRequest) (*pb.ListPeersResponse, error) {
+// ListPeers list peers
```

Comment:
I think it's better to have the original comment as its more descriptive of what this function is doing